tests/unit/test_xasprintf_x.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-FileCopyrightText: 2023, Alejandro Colomar <alx@kernel.org>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+
+#include <setjmp.h>
+#include <stddef.h>
+#include <stdlib.h>
+
+#include <stdarg.h>  // Required by <cmocka.h>
+#include <stddef.h>  // Required by <cmocka.h>
+#include <setjmp.h>  // Required by <cmocka.h>
+#include <stdint.h>  // Required by <cmocka.h>
+#include <cmocka.h>
+
+#include "sprintf.h"
+
+
+#define assert_unreachable()  assert_true(0)
+
+
+static jmp_buf  jmpb;
+
+
+/**********************
+ * WRAPPERS
+ **********************/
+int
+__wrap_vasprintf(char **restrict p, const char *restrict fmt, va_list ap)
+{
+	return -1;
+}
+
+
+void
+__wrap_exit(int status)
+{
+	longjmp(jmpb, 42);
+}
+
+
+/**********************
+ * TEST
+ **********************/
+static void
+test_xasprintf_x(void **state)
+{
+	volatile int    len;
+	char *volatile  p;
+
+	len = 0;
+
+	switch (setjmp(jmpb)) {
+	case 0:
+		len = -36;
```

Comment:
Can you change these magic values to a macro definition with a more meaningful name? The following comes to my mind:
```
#define EXIT_CALLED 42
#define XASPRINTF_CALLED -36
#define TEST_OK -6
```

Reply:
Yup, I thought about it yesterday, but felt too lazy.  Will do.  :)

Reply:
Done.